pypif/interop/mdf.py

```diff
@@ -0,0 +1,94 @@
+from pypif.util.read_view import ReadView
+from pypif.obj.common import Value
+from json import dumps
+
+
+def pif_to_mdf_record(pif):
+   """Convert a PIF into partial MDF record"""
+   res = {}
+   res["mdf"] = _to_meta_data(pif)
+   res["{source_name}"] = _to_user_defined(pif)
+   return dumps(res)
+
+
+def _to_meta_data(pif_obj):
+    """Convert the meta-data from the PIF into MDF"""
+    pif = pif_obj.as_dictionary()
+    mdf = {}
+    if pif["category"] == "system.chemical":
+        mdf["title"] = pif["names"][0]
+        mdf["composition"] = pif["chemicalFormula"]
+        mdf["acl"] = ["public"] #TODO: Real ACLs
+#        mdf["source_name"] = _construct_new_key(pif["names"])
+        mdf["data_contact"] = []
+        for contact in pif["contacts"]:
+            data_c = {
+                "given_name": contact["name"]["given"],
+                "family_name": contact["name"]["family"]
+                }
+            if contact.get("email"):
+                data_c["email"] = contact.get("email", "")
+            if contact.get("orcid"):
+                data_c["orcid"] = contact.get("orcid", "")
+            mdf["data_contact"].append(data_c)
+        mdf["data_contributor"] = [{}] #TODO: Real contrib
+        mdf["citation"] = [r["doi"] for r in pif["references"]] #TODO: Make citation
+        mdf["author"] = []
+        for ref in pif["references"]:
+            for author in ref["authors"]:
+                mdf["author"].append({
+                    "given_name": author["given"],
```

Comment:
Technically, we don't enforce that given and family names are provided, so this should be protected against them being missing.